rosbag2_cpp/src/rosbag2_cpp/message_definitions/local_message_definition_source.cpp

```diff
@@ -114,84 +121,107 @@ static std::string delimiter(const DefinitionIdentifier & definition_identifier)
     default:
       throw std::runtime_error("switch is not exhaustive");
   }
-  result += definition_identifier.package_resource_name;
+  result += definition_identifier.topic_type();
   result += "\n";
   return result;
 }
 
-MessageSpec::MessageSpec(Format format, std::string text, const std::string & package_context)
-    : dependencies(parse_dependencies(format, text, package_context))
-    , text(std::move(text))
-    , format(format)
+LocalMessageDefinitionSource::MessageSpec::MessageSpec(
+  Format format, std::string text,
+  const std::string & package_context)
+: dependencies(parse_definition_dependencies(format, text, package_context))
+  , text(std::move(text))
+  , format(format)
 {
 }
 
-const MessageSpec & MessageDefinitionCache::load_message_spec(
+const LocalMessageDefinitionSource::MessageSpec & LocalMessageDefinitionSource::load_message_spec(
   const DefinitionIdentifier & definition_identifier)
 {
   if (auto it = msg_specs_by_definition_identifier_.find(definition_identifier);
-      it != msg_specs_by_definition_identifier_.end()) {
+    it != msg_specs_by_definition_identifier_.end())
+  {
     return it->second;
   }
   std::smatch match;
-  if (!std::regex_match(definition_identifier.package_resource_name, match,
-                        PACKAGE_TYPENAME_REGEX)) {
-    throw std::invalid_argument("Invalid package resource name: " +
-                                definition_identifier.package_resource_name);
+  const auto topic_type = definition_identifier.topic_type();
+  if (!std::regex_match(topic_type, match, PACKAGE_TYPENAME_REGEX)) {
+    throw std::invalid_argument("Invalid topic_type: " + topic_type);
   }
   std::string package = match[1];
   std::string share_dir = ament_index_cpp::get_package_share_directory(package);
   std::ifstream file{share_dir + "/msg/" + match[2].str() +
-                     extension_for_format(definition_identifier.format)};
+    extension_for_format(definition_identifier.format())};
   if (!file.good()) {
-    throw DefinitionNotFoundError(definition_identifier.package_resource_name);
+    throw DefinitionNotFoundError(definition_identifier.topic_type());
   }
 
   std::string contents{std::istreambuf_iterator(file), {}};
   const MessageSpec & spec =
     msg_specs_by_definition_identifier_
-      .emplace(definition_identifier,
-               MessageSpec(definition_identifier.format, std::move(contents), package))
-      .first->second;
+    .emplace(
+    definition_identifier,
```

Comment:
Tentative suggestion for beautification after ~~uglifier~~ uncrustify
```
const MessageSpec & spec = msg_specs_by_definition_identifier_.emplace(
    definition_identifier,
    MessageSpec(definition_identifier.format(), std::move(contents), package)).first->second;
```
Note. In this form uncrustify also passes.